dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/trackedentityinstance/TrackedEntityInstanceQueryLimitTest.java

```diff
@@ -0,0 +1,215 @@
+        tei3 = createTrackedEntityInstance( orgUnitA );
+        tei4 = createTrackedEntityInstance( orgUnitA );
+        tei1.setTrackedEntityType( teiType );
+        tei2.setTrackedEntityType( teiType );
+        tei3.setTrackedEntityType( teiType );
+        tei4.setTrackedEntityType( teiType );
+
+        trackedEntityInstanceService.addTrackedEntityInstance( tei1 );
+        trackedEntityInstanceService.addTrackedEntityInstance( tei2 );
+        trackedEntityInstanceService.addTrackedEntityInstance( tei3 );
+        trackedEntityInstanceService.addTrackedEntityInstance( tei4 );
+
+        pi1 = createProgramInstance( program, tei1, orgUnitA );
+        pi2 = createProgramInstance( program, tei2, orgUnitA );
+        pi3 = createProgramInstance( program, tei3, orgUnitA );
+        pi4 = createProgramInstance( program, tei4, orgUnitA );
+
+        programInstanceService.addProgramInstance( pi1 );
+        programInstanceService.addProgramInstance( pi2 );
+        programInstanceService.addProgramInstance( pi3 );
+        programInstanceService.addProgramInstance( pi4 );
+
+        programInstanceService.enrollTrackedEntityInstance( tei1, program, new Date(), new Date(), orgUnitA );
+        programInstanceService.enrollTrackedEntityInstance( tei2, program, new Date(), new Date(), orgUnitA );
+        programInstanceService.enrollTrackedEntityInstance( tei3, program, new Date(), new Date(), orgUnitA );
+        programInstanceService.enrollTrackedEntityInstance( tei4, program, new Date(), new Date(), orgUnitA );
+
+        userService.addUser( user );
+    }
+
+    @Test
+    void testConfiguredMaxTeiLimit()
+    {
+        systemSettingManager.saveSystemSetting( SettingKey.TRACKED_ENTITY_MAX_LIMIT, 3 );
+        TrackedEntityInstanceQueryParams params = new TrackedEntityInstanceQueryParams();
+        params.setProgram( program );
+        params.setOrganisationUnits( Sets.newHashSet( orgUnitA ) );
+        params.setOrganisationUnitMode( OrganisationUnitSelectionMode.ALL );
+        params.setUser( user );
+        params.setSkipPaging( true );
+
+        List<Long> teis = trackedEntityInstanceService.getTrackedEntityInstanceIds( params,
+            false, false );
+
+        assertNotNull( teis );
+        assertEquals( 3, teis.size(), "Size cannot be more than configured Tei max limit" );
+    }
+
+    @Test
+    void testDefaultMaxTeiLimit()
+    {
+        TrackedEntityInstanceQueryParams params = new TrackedEntityInstanceQueryParams();
+        params.setProgram( program );
+        params.setOrganisationUnits( Sets.newHashSet( orgUnitA ) );
+        params.setOrganisationUnitMode( OrganisationUnitSelectionMode.ALL );
+        params.setUser( user );
+        params.setSkipPaging( true );
+
+        List<Long> teis = trackedEntityInstanceService.getTrackedEntityInstanceIds( params,
+            false, false );
+
+        assertNotNull( teis );
+        assertEquals( 4, teis.size() );
```

Comment:
assertContainsOnly already checks the size using assertEquals 😄 So please remove assertEquals.